source/go-perun/app_tutorial/client.rst

```diff
@@ -0,0 +1,329 @@
+
+            // Check that we have the correct number of participants.
+            if lcp.NumPeers() != 2 { //TODO:go-perun rename NumPeers to NumParts, Peers to Participants anywhere where all parties are referred to
+                return nil, fmt.Errorf("Invalid number of participants: %d", lcp.NumPeers())
+            }
+
+            // Check that the channel has the expected assets.
+            err := channel.AssetsAssertEqual(lcp.InitBals.Assets, []channel.Asset{c.currency})
+            if err != nil {
+                return nil, fmt.Errorf("Invalid assets: %v\n", err)
+            }
+
+            // Check that the channel has the expected assets and funding balances.
+            const assetIdx, peerIdx = 0, 1
+            if err := channel.AssetsAssertEqual(lcp.InitBals.Assets, []channel.Asset{c.currency}); err != nil {
+                return nil, fmt.Errorf("Invalid assets: %v\n", err)
+            } else if lcp.FundingAgreement[assetIdx][peerIdx].Cmp(c.stake) != 0 {
+                return nil, fmt.Errorf("Invalid funding balance")
+            }
+            return lcp, nil
+        }()
+        if err != nil {
+            r.Reject(context.TODO(), err.Error()) //nolint:errcheck // It's OK if rejection fails.
+        }
+
+As in the payment channel example, we create the accept message, send it, and start the on-chain watcher.
+
+.. code-block:: go
+
+        // Create a channel accept message and send it.
+        accept := lcp.Accept(
+            c.account,                // The account we use in the channel.
+            client.WithRandomNonce(), // Our share of the channel nonce.
+        )
+        ch, err := r.Accept(context.TODO(), accept)
+        if err != nil {
+            fmt.Printf("Error accepting channel proposal: %v\n", err)
+            return
+        }
+
+        // Start the on-chain event watcher. It automatically handles disputes.
+        c.startWatching(ch)
+
+        c.channels <- newAppChannel(ch)
+    }
+
+.. _app_client_channel:
+
+App channel
+-----------
+
+We implement the type ``AppChannel`` that wraps a Perun channel and provides convenient functions to interact with the app channel.
+We put this functionality in ``client/channel.go``.
+
+.. code-block:: go
+
+    type AppChannel struct {
+        ch *client.Channel
+    }
+
+.. _app_client_channel_set:
+
+Set
+...
+We create ``Set``, which expects ``x`` and ``y`` that indicate where the client wants to put its symbol on the grid.
+
+We use ``channel.UpdateBy`` for conveniently proposing our desired update to the channel's ``state``.
+With ``state.App``, we fetch the app that identifies the channel's application and check if it is of the expected type.
+Then we call ``TicTacToeApp.Set``, which will manipulate ``state.Data`` to include the desired turn for us.
+``state.Data`` holds all app-specific data, reflecting our Tic-Tac-Toe game's current game state.
+We will go into more detail about this at :ref:`a later stage <app_set>`.
```

Comment:
```suggestion
We will go into more detail about this in the :ref:`app description <app_set>`.
```

Try to avoid relative references like "later" or "earlier", as they assume a certain structure. Better use absolute references.